cranelift/codegen/src/machinst/buffer.rs

```diff
@@ -334,7 +334,7 @@ impl<I: VCodeInst> MachBuffer<I> {
     /// times, e.g. after calling `add_{cond,uncond}_branch()` and
     /// before emitting branch bytes.
     fn check_label_branch_invariants(&self) {
-        if !cfg!(debug_assertions) || cfg!(fuzzing) {
+        if !cfg!(fuzzing) {
```

Comment:
Can you keep this when debug assertions are enabled?

Reply:
This actually became a significant O(n^2) slowdown in debug builds because of the additional edge blocks introduced in the block-lowering order, and was causing some test timeouts. IMHO we've tested and relied on this code enough over the past almost-2-years that enabling the checks only in fuzzing and not in all debug builds is probably OK...

Reply:
FWIW, we see ~60x slowdown when fuzzing and only have ~1 minute per fuzz input (so roughly 1s budget on non-fuzz builds), so it seems semi-likely that these checks will be too expensive for fuzzing as well if they are too expensive for `debug_assert!`.